lib/src/repo.rs

```diff
@@ -1082,22 +1081,27 @@ impl MutableRepo {
         &mut self,
         settings: &UserSettings,
         old_commit_id: &CommitId,
-        new_commit_id: &CommitId,
+        new_commit_ids: &[CommitId],
         abandoned_old_commit: bool,
     ) -> BackendResult<()> {
         let workspaces_to_update = self.view().workspaces_for_wc_commit_id(old_commit_id);
         if workspaces_to_update.is_empty() {
             return Ok(());
         }
 
-        let new_commit = self.store().get_commit(new_commit_id)?;
         let new_wc_commit = if !abandoned_old_commit {
-            new_commit
+            // We arbitrarily pick a new working-copy commit among the candidates.
+            self.store().get_commit(&new_commit_ids[0])?
```

Comment:
(no action needed) I assume this would be a conflict ref (if we had such thing.)